Northstar.CustomServers/mod/scripts/vscripts/sh_progression.nut

```diff
@@ -84,6 +85,29 @@ bool function ClientCommand_SetProgression( entity player, array<string> args )
 
 	return true
 }
+
+bool function ClientCommand_ResetTitanAegis( entity player, array<string> args )
+{
+	int suitIndex = GetPersistentSpawnLoadoutIndex( player, "titan" )
+	player.SetPersistentVar( "titanFDUnlockPoints[" + suitIndex + "]", 0 )
+	player.SetPersistentVar( "previousFDUnlockPoints[" + suitIndex + "]", 0 )
+	player.SetPersistentVar( "fdTitanXP[" + suitIndex + "]", 0 )
+	player.SetPersistentVar( "fdPreviousTitanXP[" + suitIndex + "]", 0 )
+	
+	//Refresh Highest Aegis Titan since we might get all of them back to 1 if players wants
+	int enumCount = PersistenceGetEnumCount( "titanClasses" )
+	int HighestAegis = 0
+	for ( int i = 0; i < enumCount; i++ )
+	{
+		string enumName = PersistenceGetEnumItemNameForIndex( "titanClasses", i )
+		int AegisLevel = FD_TitanGetLevelForXP( enumName, FD_TitanGetXP( player, enumName ) )
+		if ( HighestAegis < AegisLevel )
+			HighestAegis = AegisLevel
+	}
+	player.SetPersistentVar( "fdStats.highestTitanFDLevel", HighestAegis )
```

Comment:
Is this duplicated code from somewhere? Even if not, having this as a util function would be nice later (something like `RecalculateHighestTitanFDLevel` maybe)

Reply:
Hmm great idea, i haven't thought of that, and yeah code is not duplicated, no vanilla function that does this from what i remember.